x/mongo/driver/errors.go

```diff
@@ -48,6 +48,8 @@ var (
 	// ErrDeadlineWouldBeExceeded is returned when a Timeout set on an operation would be exceeded
 	// if the operation were sent to the server.
 	ErrDeadlineWouldBeExceeded = errors.New("operation not sent to server, as Timeout would be exceeded")
+	// ErrNegativeMaxTime is returned when MaxTime on an operation is a negative value.
+	ErrNegativeMaxTime = errors.New("a negative value was provided for MaxTime on an operation")
```

Comment:
I've added a new error value for when users provide negative `MaxTime` values in their operation options. I think this is better than passing the negative (non-valid) value to the server and letting the server error, or silently discarding the value on our side.